diff -ur deepin-system-monitor-5.6.3.1.old/src/compact_cpu_monitor.cpp deepin-system-monitor-5.6.3.1/src/compact_cpu_monitor.cpp
--- deepin-system-monitor-5.6.3.1.old/src/compact_cpu_monitor.cpp	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/compact_cpu_monitor.cpp	2020-06-05 14:54:02.193021607 +0800
@@ -21,6 +21,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <QPainterPath>
 #include <DApplication>
 #include <DApplicationHelper>
 #include <DHiDPIHelper>
diff -ur deepin-system-monitor-5.6.3.1.old/src/compact_disk_monitor.cpp deepin-system-monitor-5.6.3.1/src/compact_disk_monitor.cpp
--- deepin-system-monitor-5.6.3.1.old/src/compact_disk_monitor.cpp	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/compact_disk_monitor.cpp	2020-06-05 14:54:02.193021607 +0800
@@ -21,6 +21,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <QPainterPath>
 #include <DApplication>
 #include <DApplicationHelper>
 #include <DHiDPIHelper>
diff -ur deepin-system-monitor-5.6.3.1.old/src/compact_disk_monitor.h deepin-system-monitor-5.6.3.1/src/compact_disk_monitor.h
--- deepin-system-monitor-5.6.3.1.old/src/compact_disk_monitor.h	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/compact_disk_monitor.h	2020-06-05 14:54:02.193021607 +0800
@@ -24,6 +24,7 @@
 #ifndef COMPACTDISKMONITOR_H
 #define COMPACTDISKMONITOR_H
 
+#include <QPainterPath>
 #include <DApplicationHelper>
 #include <QWidget>
 
diff -ur deepin-system-monitor-5.6.3.1.old/src/compact_memory_monitor.cpp deepin-system-monitor-5.6.3.1/src/compact_memory_monitor.cpp
--- deepin-system-monitor-5.6.3.1.old/src/compact_memory_monitor.cpp	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/compact_memory_monitor.cpp	2020-06-05 14:54:02.193021607 +0800
@@ -21,6 +21,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <QPainterPath>
 #include <DApplication>
 #include <DApplicationHelper>
 #include <DHiDPIHelper>
diff -ur deepin-system-monitor-5.6.3.1.old/src/compact_network_monitor.cpp deepin-system-monitor-5.6.3.1/src/compact_network_monitor.cpp
--- deepin-system-monitor-5.6.3.1.old/src/compact_network_monitor.cpp	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/compact_network_monitor.cpp	2020-06-05 14:54:02.193021607 +0800
@@ -21,6 +21,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <QPainterPath>
 #include <DApplication>
 #include <DApplicationHelper>
 #include <DHiDPIHelper>
diff -ur deepin-system-monitor-5.6.3.1.old/src/compact_network_monitor.h deepin-system-monitor-5.6.3.1/src/compact_network_monitor.h
--- deepin-system-monitor-5.6.3.1.old/src/compact_network_monitor.h	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/compact_network_monitor.h	2020-06-05 14:54:02.193021607 +0800
@@ -24,6 +24,7 @@
 #ifndef COMPACTNETWORKMONITOR_H
 #define COMPACTNETWORKMONITOR_H
 
+#include <QPainterPath>
 #include <DApplicationHelper>
 #include <QWidget>
 
diff -ur deepin-system-monitor-5.6.3.1.old/src/cpu_monitor.cpp deepin-system-monitor-5.6.3.1/src/cpu_monitor.cpp
--- deepin-system-monitor-5.6.3.1.old/src/cpu_monitor.cpp	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/cpu_monitor.cpp	2020-06-05 14:54:02.203039059 +0800
@@ -21,6 +21,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <QPainterPath>
 #include <DApplication>
 #include <DApplicationHelper>
 #include <DHiDPIHelper>
diff -ur deepin-system-monitor-5.6.3.1.old/src/cpu_monitor.h deepin-system-monitor-5.6.3.1/src/cpu_monitor.h
--- deepin-system-monitor-5.6.3.1.old/src/cpu_monitor.h	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/cpu_monitor.h	2020-06-05 14:54:02.193021607 +0800
@@ -24,6 +24,7 @@
 #ifndef CpuMONITOR_H
 #define CpuMONITOR_H
 
+#include <QPainterPath>
 #include <DApplicationHelper>
 #include <QIcon>
 #include <QList>
diff -ur deepin-system-monitor-5.6.3.1.old/src/disk_monitor.cpp deepin-system-monitor-5.6.3.1/src/disk_monitor.cpp
--- deepin-system-monitor-5.6.3.1.old/src/disk_monitor.cpp	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/disk_monitor.cpp	2020-06-05 14:54:02.193021607 +0800
@@ -21,6 +21,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <QPainterPath>
 #include <DApplication>
 #include <DApplicationHelper>
 #include <DHiDPIHelper>
diff -ur deepin-system-monitor-5.6.3.1.old/src/disk_monitor.h deepin-system-monitor-5.6.3.1/src/disk_monitor.h
--- deepin-system-monitor-5.6.3.1.old/src/disk_monitor.h	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/disk_monitor.h	2020-06-05 14:54:02.193021607 +0800
@@ -24,6 +24,7 @@
 #ifndef DISKMONITOR_H
 #define DISKMONITOR_H
 
+#include <QPainterPath>
 #include <DApplicationHelper>
 #include <QWidget>
 
diff -ur deepin-system-monitor-5.6.3.1.old/src/gui/base_header_view.cpp deepin-system-monitor-5.6.3.1/src/gui/base_header_view.cpp
--- deepin-system-monitor-5.6.3.1.old/src/gui/base_header_view.cpp	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/gui/base_header_view.cpp	2020-06-05 14:54:02.193021607 +0800
@@ -1,3 +1,4 @@
+#include <QPainterPath>
 #include <cmath>
 
 #include <DApplication>
diff -ur deepin-system-monitor-5.6.3.1.old/src/gui/base_table_view.cpp deepin-system-monitor-5.6.3.1/src/gui/base_table_view.cpp
--- deepin-system-monitor-5.6.3.1.old/src/gui/base_table_view.cpp	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/gui/base_table_view.cpp	2020-06-05 14:54:02.203039059 +0800
@@ -1,3 +1,4 @@
+#include <QPainterPath>
 #include <DApplication>
 #include <DApplicationHelper>
 #include <DPalette>
diff -ur deepin-system-monitor-5.6.3.1.old/src/gui/process_page_widget.cpp deepin-system-monitor-5.6.3.1/src/gui/process_page_widget.cpp
--- deepin-system-monitor-5.6.3.1.old/src/gui/process_page_widget.cpp	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/gui/process_page_widget.cpp	2020-06-05 14:54:02.193021607 +0800
@@ -21,6 +21,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <QPainterPath>
 #include <signal.h>
 #include <iostream>
 
diff -ur deepin-system-monitor-5.6.3.1.old/src/gui/system_service_page_widget.cpp deepin-system-monitor-5.6.3.1/src/gui/system_service_page_widget.cpp
--- deepin-system-monitor-5.6.3.1.old/src/gui/system_service_page_widget.cpp	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/gui/system_service_page_widget.cpp	2020-06-05 14:54:02.203039059 +0800
@@ -1,3 +1,4 @@
+#include <QPainterPath>
 #include <DApplication>
 #include <DApplicationHelper>
 #include <DStyle>
diff -ur deepin-system-monitor-5.6.3.1.old/src/memory_monitor.cpp deepin-system-monitor-5.6.3.1/src/memory_monitor.cpp
--- deepin-system-monitor-5.6.3.1.old/src/memory_monitor.cpp	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/memory_monitor.cpp	2020-06-05 14:54:02.193021607 +0800
@@ -21,6 +21,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <QPainterPath>
 #include <DApplication>
 #include <DApplicationHelper>
 #include <DHiDPIHelper>
diff -ur deepin-system-monitor-5.6.3.1.old/src/network_monitor.cpp deepin-system-monitor-5.6.3.1/src/network_monitor.cpp
--- deepin-system-monitor-5.6.3.1.old/src/network_monitor.cpp	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/network_monitor.cpp	2020-06-05 14:54:02.193021607 +0800
@@ -21,6 +21,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <QPainterPath>
 #include <DApplication>
 #include <DApplicationHelper>
 #include <DHiDPIHelper>
diff -ur deepin-system-monitor-5.6.3.1.old/src/network_monitor.h deepin-system-monitor-5.6.3.1/src/network_monitor.h
--- deepin-system-monitor-5.6.3.1.old/src/network_monitor.h	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/network_monitor.h	2020-06-05 14:54:02.193021607 +0800
@@ -24,6 +24,7 @@
 #ifndef NETWORKMONITOR_H
 #define NETWORKMONITOR_H
 
+#include <QPainterPath>
 #include <DApplicationHelper>
 #include <QIcon>
 #include <QWidget>
diff -ur deepin-system-monitor-5.6.3.1.old/src/utils.cpp deepin-system-monitor-5.6.3.1/src/utils.cpp
--- deepin-system-monitor-5.6.3.1.old/src/utils.cpp	2020-06-05 14:53:56.028073166 +0800
+++ deepin-system-monitor-5.6.3.1/src/utils.cpp	2020-06-05 14:54:02.203039059 +0800
@@ -21,6 +21,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <QPainterPath>
 #include <stdio.h>
 #include <time.h>
 #include <fstream>
