diff --git a/3rdparty/terminalwidget/lib/SessionManager.h b/3rdparty/terminalwidget/lib/SessionManager.h
index 53c142c..ace7e20 100644
--- a/3rdparty/terminalwidget/lib/SessionManager.h
+++ b/3rdparty/terminalwidget/lib/SessionManager.h
@@ -26,6 +26,8 @@
 // Qt
 #include <QHash>
 #include <QList>
+#include <QObject>
+#include <QMap>
 
 namespace Konsole {
 class Session;
diff --git a/src/customcommand/customcommanddelegate.cpp b/src/customcommand/customcommanddelegate.cpp
index ed3d156..bf55cda 100644
--- a/src/customcommand/customcommanddelegate.cpp
+++ b/src/customcommand/customcommanddelegate.cpp
@@ -2,6 +2,7 @@
 #include "customcommanditemmodel.h"
 #include "utils.h"
 
+#include <QPainterPath>
 #include <DGuiApplicationHelper>
 #include <DApplicationHelper>
 #include <DPalette>
diff --git a/src/remotemanage/serverconfigdelegate.cpp b/src/remotemanage/serverconfigdelegate.cpp
index 8d574fe..b8ce5c7 100644
--- a/src/remotemanage/serverconfigdelegate.cpp
+++ b/src/remotemanage/serverconfigdelegate.cpp
@@ -2,6 +2,7 @@
 #include "serverconfigitemmodel.h"
 #include "utils.h"
 
+#include <QPainterPath>
 #include <DApplicationHelper>
 #include <DPalette>
 #include <DStyleHelper>
diff --git a/src/theme/themeitemdelegate.cpp b/src/theme/themeitemdelegate.cpp
index 92bc9be..2726440 100644
--- a/src/theme/themeitemdelegate.cpp
+++ b/src/theme/themeitemdelegate.cpp
@@ -1,5 +1,6 @@
 #include "themeitemdelegate.h"
 
+#include <QPainterPath>
 #include <QPainter>
 
 ThemeItemDelegate::ThemeItemDelegate(QAbstractItemView *parent) : DStyledItemDelegate(parent)
