--- a/application/common/collator.cpp	2020-03-11 14:33:40.000000000 +0800
+++ b/application/common/collator.cpp	2020-04-19 17:44:49.829336869 +0800
@@ -37,9 +37,10 @@
     UErrorCode ec = U_ZERO_ERROR;
     m_collator = icu::Collator::createInstance(ec);
     if (U_FAILURE(ec)) {
-#undef u_errorName
-#define UERRORNAME U_DEF2_ICU_ENTRY_POINT_RENAME(u_errorName, U_ICU_VERSION_SUFFIX)
-        qDebug() << QString("Create collator failed: [%1] %2").arg(ec).arg(UERRORNAME(ec));
+//#undef u_errorName
+//#define UERRORNAME U_DEF2_ICU_ENTRY_POINT_RENAME(u_errorName, U_ICU_VERSION_SUFFIX)
+        //qDebug() << QString("Create collator failed: [%1] %2").arg(ec).arg(UERRORNAME(ec));
+        qDebug() << QString("Create collator failed: [%1] %2").arg(ec);
         m_collator = nullptr;
     }
 }
