--- a/application/application.pro
+++ b/application/application.pro
@@ -6,11 +6,9 @@ TARGET = deepin-log-viewer
 TEMPLATE = app
 CONFIG += c++11 link_pkgconfig
 
-LIBS += -lsystemd
-
 DEFINES += USE_POLKIT ENABLE_INACTIVE_DISPLAY
 
-LIBS += -lsystemd -licui18n -licuuc
+LIBS += -lelogind -licui18n -licuuc
 
 include(../3rdparty/QtXlsxWriter/src/xlsx/qtxlsx.pri)
 include(../3rdparty/docx/docx.pri)

