diff -ur deepin-reader-5.6.7.old/src/CustomControl/RoundColorWidget.cpp deepin-reader-5.6.7/src/CustomControl/RoundColorWidget.cpp
--- deepin-reader-5.6.7.old/src/CustomControl/RoundColorWidget.cpp	2020-06-05 14:41:10.216415848 +0800
+++ deepin-reader-5.6.7/src/CustomControl/RoundColorWidget.cpp	2020-06-05 14:41:16.917701226 +0800
@@ -20,6 +20,7 @@
  */
 #include "RoundColorWidget.h"
 
+#include <QPainterPath>
 #include <DStyle>
 
 #include <QColor>
diff -ur deepin-reader-5.6.7.old/src/djvuControl/SheetBrowserDJVU.cpp deepin-reader-5.6.7/src/djvuControl/SheetBrowserDJVU.cpp
--- deepin-reader-5.6.7.old/src/djvuControl/SheetBrowserDJVU.cpp	2020-06-05 14:41:10.216415848 +0800
+++ deepin-reader-5.6.7/src/djvuControl/SheetBrowserDJVU.cpp	2020-06-05 14:44:44.521212137 +0800
@@ -3,6 +3,10 @@
 #include "SheetBrowserDJVUItem.h"
 #include "DocSheetDJVU.h"
 
+#include <QPainterPath>
+#include <QWheelEvent>
+#include <QMouseEvent>
+#include <QDragEnterEvent>
 #include <QDebug>
 #include <QGraphicsItem>
 #include <QScrollBar>
diff -ur deepin-reader-5.6.7.old/src/document/djvumodel.cpp deepin-reader-5.6.7/src/document/djvumodel.cpp
--- deepin-reader-5.6.7.old/src/document/djvumodel.cpp	2020-06-05 14:41:10.216415848 +0800
+++ deepin-reader-5.6.7/src/document/djvumodel.cpp	2020-06-05 14:41:16.917701226 +0800
@@ -1,5 +1,6 @@
 #include "djvumodel.h"
 
+#include <QPainterPath>
 #include <cstdio>
 #include <QDebug>
 #include <QFile>
diff -ur deepin-reader-5.6.7.old/src/document/model.h deepin-reader-5.6.7/src/document/model.h
--- deepin-reader-5.6.7.old/src/document/model.h	2020-06-05 14:41:10.216415848 +0800
+++ deepin-reader-5.6.7/src/document/model.h	2020-06-05 14:41:16.917701226 +0800
@@ -1,6 +1,7 @@
 #ifndef DOCUMENTMODEL_H
 #define DOCUMENTMODEL_H
 
+#include <QPainterPath>
 #include <QList>
 #include <QtPlugin>
 #include <QRect>
diff -ur deepin-reader-5.6.7.old/src/lpreviewControl/bookmark/bookmarkdelegate.cpp deepin-reader-5.6.7/src/lpreviewControl/bookmark/bookmarkdelegate.cpp
--- deepin-reader-5.6.7.old/src/lpreviewControl/bookmark/bookmarkdelegate.cpp	2020-06-05 14:41:10.216415848 +0800
+++ deepin-reader-5.6.7/src/lpreviewControl/bookmark/bookmarkdelegate.cpp	2020-06-05 14:41:16.917701226 +0800
@@ -17,6 +17,7 @@
 #include "bookmarkdelegate.h"
 #include "lpreviewControl/ImageViewModel.h"
 
+#include <QPainterPath>
 #include <QPainter>
 #include <QDebug>
 #include <QItemSelectionModel>
diff -ur deepin-reader-5.6.7.old/src/lpreviewControl/note/notesdelegate.cpp deepin-reader-5.6.7/src/lpreviewControl/note/notesdelegate.cpp
--- deepin-reader-5.6.7.old/src/lpreviewControl/note/notesdelegate.cpp	2020-06-05 14:41:10.216415848 +0800
+++ deepin-reader-5.6.7/src/lpreviewControl/note/notesdelegate.cpp	2020-06-05 14:41:16.917701226 +0800
@@ -18,6 +18,7 @@
 #include "lpreviewControl/ImageViewModel.h"
 #include "utils/utils.h"
 
+#include <QPainterPath>
 #include <QPainter>
 #include <QDebug>
 #include <QItemSelectionModel>
diff -ur deepin-reader-5.6.7.old/src/lpreviewControl/search/searchresdelegate.cpp deepin-reader-5.6.7/src/lpreviewControl/search/searchresdelegate.cpp
--- deepin-reader-5.6.7.old/src/lpreviewControl/search/searchresdelegate.cpp	2020-06-05 14:41:10.216415848 +0800
+++ deepin-reader-5.6.7/src/lpreviewControl/search/searchresdelegate.cpp	2020-06-05 14:41:16.917701226 +0800
@@ -18,6 +18,7 @@
 #include "lpreviewControl/ImageViewModel.h"
 #include "utils/utils.h"
 
+#include <QPainterPath>
 #include <QPainter>
 #include <QDebug>
 #include <QItemSelectionModel>
diff -ur deepin-reader-5.6.7.old/src/lpreviewControl/thumbnail/ThumbnailDelegate.cpp deepin-reader-5.6.7/src/lpreviewControl/thumbnail/ThumbnailDelegate.cpp
--- deepin-reader-5.6.7.old/src/lpreviewControl/thumbnail/ThumbnailDelegate.cpp	2020-06-05 14:41:10.216415848 +0800
+++ deepin-reader-5.6.7/src/lpreviewControl/thumbnail/ThumbnailDelegate.cpp	2020-06-05 14:41:16.917701226 +0800
@@ -18,6 +18,7 @@
 #include "lpreviewControl/ImageViewModel.h"
 #include "utils/utils.h"
 
+#include <QPainterPath>
 #include <QPainter>
 #include <QDebug>
 #include <QItemSelectionModel>
diff -ur deepin-reader-5.6.7.old/src/utils/utils.cpp deepin-reader-5.6.7/src/utils/utils.cpp
--- deepin-reader-5.6.7.old/src/utils/utils.cpp	2020-06-05 14:41:10.216415848 +0800
+++ deepin-reader-5.6.7/src/utils/utils.cpp	2020-06-05 14:41:16.917701226 +0800
@@ -22,6 +22,7 @@
 
 #include "utils.h"
 
+#include <QPainterPath>
 #include <DSettings>
 #include <DSettingsOption>
 #include <QRegularExpression>
diff -ur deepin-reader-5.6.7.old/src/widgets/SlideWidget.cpp deepin-reader-5.6.7/src/widgets/SlideWidget.cpp
--- deepin-reader-5.6.7.old/src/widgets/SlideWidget.cpp	2020-06-05 14:41:10.216415848 +0800
+++ deepin-reader-5.6.7/src/widgets/SlideWidget.cpp	2020-06-05 14:42:44.908824153 +0800
@@ -23,6 +23,7 @@
 #include <DApplication>
 
 #include <QDebug>
+#include <QMouseEvent>
 #include <QDesktopWidget>
 #include <QPropertyAnimation>
 #include <QTimer>
