diff --git a/plugins/kwin-xcb/lib/kwinutils.cpp b/plugins/kwin-xcb/lib/kwinutils.cpp
index ed68ec8..a80d964 100644
--- a/plugins/kwin-xcb/lib/kwinutils.cpp
+++ b/plugins/kwin-xcb/lib/kwinutils.cpp
@@ -589,7 +589,11 @@ QObject *KWinUtils::tabBox()
 
 QObject *KWinUtils::cursor()
 {
+#if KWIN_VERSION_MIN <= 18
     return KWin::Cursor::s_self;
+#else
+    return nullptr;
+#endif
 }
 
 QObject *KWinUtils::virtualDesktop()
diff --git a/plugins/kwineffects/blur/blur.cpp b/plugins/kwineffects/blur/blur.cpp
index b5cae9e..e3bc1eb 100644
--- a/plugins/kwineffects/blur/blur.cpp
+++ b/plugins/kwineffects/blur/blur.cpp
@@ -32,10 +32,17 @@
 #include <cmath> // for ceil()
 #include <QTimer>
 
+#if KWIN_VERSION_MIN <= 18
 #include <KWayland/Server/surface_interface.h>
 #include <KWayland/Server/blur_interface.h>
 #include <KWayland/Server/shadow_interface.h>
 #include <KWayland/Server/display.h>
+#else
+#include <KWaylandServer/surface_interface.h>
+#include <KWaylandServer/blur_interface.h>
+#include <KWaylandServer/shadow_interface.h>
+#include <KWaylandServer/display.h>
+#endif
 
 static const QByteArray s_blurAtomName = QByteArrayLiteral("_KDE_NET_WM_BLUR_BEHIND_REGION");
 
@@ -61,10 +68,16 @@ BlurEffect::BlurEffect(QObject *, const QVariantList &)
     //     Should be included in _NET_SUPPORTED instead.
     if (m_shader && m_shader->isValid() && m_renderTargetsValid) {
         net_wm_blur_region = effects->announceSupportProperty(s_blurAtomName, this);
+#if KWIN_VERSION_MIN <= 18
         KWayland::Server::Display *display = effects->waylandDisplay();
+#else
+        KWaylandServer::Display *display = effects->waylandDisplay();
+#endif
         if (display) {
             m_blurManager = display->createBlurManager(this);
+#if KWIN_VERSION_MIN <= 18
             m_blurManager->create();
+#endif
         }
     } else {
         net_wm_blur_region = 0;
@@ -310,7 +323,11 @@ void BlurEffect::updateBlurRegion(EffectWindow *w) const
         }
     }
 
+#if KWIN_VERSION_MIN <= 18
     KWayland::Server::SurfaceInterface *surf = w->surface();
+#else
+    KWaylandServer::SurfaceInterface *surf = w->surface();
+#endif
 
     if (surf && surf->blur()) {
         region = surf->blur()->region();
@@ -338,10 +355,18 @@ void BlurEffect::updateBlurRegion(EffectWindow *w) const
 
 void BlurEffect::slotWindowAdded(EffectWindow *w)
 {
+#if KWIN_VERSION_MIN <= 18
     KWayland::Server::SurfaceInterface *surf = w->surface();
+#else
+    KWaylandServer::SurfaceInterface *surf = w->surface();
+#endif
 
     if (surf) {
+#if KWIN_VERSION_MIN <= 18
         windowBlurChangedConnections[w] = connect(surf, &KWayland::Server::SurfaceInterface::blurChanged, this, [this, w] () {
+#else
+        windowBlurChangedConnections[w] = connect(surf, &KWaylandServer::SurfaceInterface::blurChanged, this, [this, w] () {
+#endif
             if (w) {
                 updateBlurRegion(w);
             }
@@ -445,7 +470,11 @@ QRegion BlurEffect::blurRegion(const EffectWindow *w) const
         const QRegion appRegion = qvariant_cast<QRegion>(value);
         if (!appRegion.isEmpty()) {
             if (w->decorationHasAlpha() && effects->decorationSupportsBlurBehind()) {
+#if KWIN_VERSION_MIN <= 18
                 region = w->shape();
+#else
+                region = w->shape() & w->rect();
+#endif
                 region -= w->decorationInnerRect();
             }
             region |= appRegion.translated(w->contentsRect().topLeft()) &
@@ -453,12 +482,20 @@ QRegion BlurEffect::blurRegion(const EffectWindow *w) const
         } else {
             // An empty region means that the blur effect should be enabled
             // for the whole window.
+#if KWIN_VERSION_MIN <= 18
             region = w->shape();
+#else
+            region = w->shape() & w->rect();
+#endif
         }
     } else if (w->decorationHasAlpha() && effects->decorationSupportsBlurBehind()) {
         // If the client hasn't specified a blur region, we'll only enable
         // the effect behind the decoration.
+#if KWIN_VERSION_MIN <= 18
         region = w->shape();
+#else
+        region = w->shape() & w->rect();
+#endif
         region -= w->decorationInnerRect();
     }
 
diff --git a/plugins/kwineffects/blur/blur.h b/plugins/kwineffects/blur/blur.h
index 7c76a82..042f093 100644
--- a/plugins/kwineffects/blur/blur.h
+++ b/plugins/kwineffects/blur/blur.h
@@ -31,6 +31,7 @@
 
 using namespace KWin;
 
+#if KWIN_VERSION_MIN <= 18
 namespace KWayland
 {
 namespace Server
@@ -38,6 +39,12 @@ namespace Server
 class BlurManagerInterface;
 }
 }
+#else
+namespace KWaylandServer
+{
+class BlurManagerInterface;
+}
+#endif
 
 static const int borderSize = 5;
 
@@ -131,7 +138,11 @@ private:
     QVector <BlurValuesStruct> blurStrengthValues;
 
     QMap <EffectWindow*, QMetaObject::Connection> windowBlurChangedConnections;
+#if KWIN_VERSION_MIN <= 18
     KWayland::Server::BlurManagerInterface *m_blurManager = nullptr;
+#else
+    KWaylandServer::BlurManagerInterface *m_blurManager = nullptr;
+#endif
 };
 
 inline
