diff --git a/viewer/frame/extensionpanel.cpp b/viewer/frame/extensionpanel.cpp
index c68d8c4..63cbb9a 100755
--- a/viewer/frame/extensionpanel.cpp
+++ b/viewer/frame/extensionpanel.cpp
@@ -15,6 +15,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 #include "extensionpanel.h"
+#include <QPainterPath>
 #include <QPainter>
 #include "application.h"
 #include "controller/signalmanager.h"
diff --git a/viewer/frame/toptoolbar.cpp b/viewer/frame/toptoolbar.cpp
index 0ca73c2..41f6df3 100755
--- a/viewer/frame/toptoolbar.cpp
+++ b/viewer/frame/toptoolbar.cpp
@@ -24,6 +24,7 @@
 #include "utils/baseutils.h"
 #include "utils/shortcut.h"
 
+#include <QPainterPath>
 #include <dtitlebar.h>
 #include <dwindowclosebutton.h>
 #include <dwindowmaxbutton.h>
diff --git a/viewer/module/view/contents/imageinfowidget.cpp b/viewer/module/view/contents/imageinfowidget.cpp
index f10b9d7..641452c 100644
--- a/viewer/module/view/contents/imageinfowidget.cpp
+++ b/viewer/module/view/contents/imageinfowidget.cpp
@@ -20,6 +20,7 @@
 #include "utils/imageutils.h"
 #include "widgets/formlabel.h"
 
+#include <QPainterPath>
 #include <DApplicationHelper>
 #include <DArrowLineDrawer>
 #include <DDialogCloseButton>
diff --git a/viewer/module/view/contents/ttbcontent.cpp b/viewer/module/view/contents/ttbcontent.cpp
index a639baa..dbdfa71 100644
--- a/viewer/module/view/contents/ttbcontent.cpp
+++ b/viewer/module/view/contents/ttbcontent.cpp
@@ -26,6 +26,7 @@
 #include "widgets/pushbutton.h"
 #include "widgets/returnbutton.h"
 
+#include <QPainterPath>
 #include <DImageButton>
 #include <DSpinner>
 #include <DThumbnailProvider>
diff --git a/viewer/settings/shortcut/shortcuteditor.cpp b/viewer/settings/shortcut/shortcuteditor.cpp
index ee4e20a..bee9efc 100644
--- a/viewer/settings/shortcut/shortcuteditor.cpp
+++ b/viewer/settings/shortcut/shortcuteditor.cpp
@@ -17,6 +17,7 @@
 #include "shortcuteditor.h"
 #include "application.h"
 #include "controller/configsetter.h"
+#include <QPainterPath>
 #include <QFocusEvent>
 #include <QFontMetrics>
 #include <QPainter>
diff --git a/viewer/settings/titlebutton.cpp b/viewer/settings/titlebutton.cpp
index a9de26b..c74c8a5 100755
--- a/viewer/settings/titlebutton.cpp
+++ b/viewer/settings/titlebutton.cpp
@@ -15,6 +15,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 #include "titlebutton.h"
+#include <QPainterPath>
 #include <QMouseEvent>
 #include <QPainter>
 #include <QPainterPath>
diff --git a/viewer/widgets/blureframe.cpp b/viewer/widgets/blureframe.cpp
index 3ad7ef3..33b8311 100755
--- a/viewer/widgets/blureframe.cpp
+++ b/viewer/widgets/blureframe.cpp
@@ -16,6 +16,7 @@
  */
 #include "blureframe.h"
 #include "utils/baseutils.h"
+#include <QPainterPath>
 #include <QApplication>
 #include <QDebug>
 #include <QPainter>
diff --git a/viewer/widgets/dialogs/imginfodialog.cpp b/viewer/widgets/dialogs/imginfodialog.cpp
index 65f6853..01eb472 100755
--- a/viewer/widgets/dialogs/imginfodialog.cpp
+++ b/viewer/widgets/dialogs/imginfodialog.cpp
@@ -21,6 +21,7 @@
 #include "../imagebutton.h"
 #include "widgets/formlabel.h"
 
+#include <QPainterPath>
 #include <dtitlebar.h>
 #include <QFormLayout>
 #include <QKeyEvent>
diff --git a/viewer/widgets/popupmenustyle.cpp b/viewer/widgets/popupmenustyle.cpp
index ddb509c..8375155 100755
--- a/viewer/widgets/popupmenustyle.cpp
+++ b/viewer/widgets/popupmenustyle.cpp
@@ -15,6 +15,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 #include "popupmenustyle.h"
+#include <QPainterPath>
 #include <QDebug>
 #include <QPixmap>
 #include <QPixmapCache>
diff --git a/viewer/widgets/thumbnaildelegate.cpp b/viewer/widgets/thumbnaildelegate.cpp
index 64bd65e..a384214 100755
--- a/viewer/widgets/thumbnaildelegate.cpp
+++ b/viewer/widgets/thumbnaildelegate.cpp
@@ -18,6 +18,7 @@
 #include "application.h"
 #include "utils/imageutils.h"
 #include "utils/baseutils.h"
+#include <QPainterPath>
 #include <QDateTime>
 #include <QDebug>
 #include <QHBoxLayout>
diff --git a/viewer/widgets/thumbnaillistview.cpp b/viewer/widgets/thumbnaillistview.cpp
index 1f52174..da17b60 100755
--- a/viewer/widgets/thumbnaillistview.cpp
+++ b/viewer/widgets/thumbnaillistview.cpp
@@ -23,6 +23,7 @@
 #include "utils/baseutils.h"
 #include "utils/imageutils.h"
 #include "widgets/scrollbar.h"
+#include <QPainterPath>
 #include <QBuffer>
 #include <QDebug>
 #include <QEvent>
